public/app/features/dashboard/components/DashNav/DashNav.tsx

```diff
@@ -321,6 +322,10 @@ export const DashNav = React.memo<Props>((props) => {
       return [renderPlaylistControls(), renderTimeControls()];
     }
 
+    if (props.isPublic && !!props.dashboard.meta.publicDashboardTimeSelectionEnabled) {
+      return [renderTimeControls()];
+    }
```

Comment:
@juanicabanas for example both these conditionals can be removed, and you can just use the dashboard model's hide hideTimePicker option instead `dashboard.timepicker.hidden`  (set in backend to false)

Reply:
@juanicabanas and this condition /option is already checked inside renderTimeControls